core/src/test/java/hudson/diagnosis/ReverseProxySetupMonitorTest.java

```diff
@@ -0,0 +1,54 @@
+package hudson.diagnosis;
+
+import jenkins.model.Jenkins;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kohsuke.stapler.HttpResponses;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.*;
+import static org.mockito.BDDMockito.given;
+
+/**
+ * Created by haswell on 8/11/17.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class ReverseProxySetupMonitorTest {
+  
+    private ReverseProxySetupMonitor proxySetupMonitor;
+    
+    @Mock
+    private Jenkins jenkins;
+    
+    
+    @Before
+    public void setUp() {
+        proxySetupMonitor = new ReverseProxySetupMonitor();
+    }
+    
+
+    @Test
+    public void ensureSplittingIsCorrect() throws UnsupportedEncodingException {
+        given(jenkins.getRootUrl()).willReturn("http://localhost:8080");
```

Comment:
I'll change that, but this is just to check the referrer construction.  None of the system tests will test this since this is not manifested in Jetty.